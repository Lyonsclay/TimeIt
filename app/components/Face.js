import React from 'react';
import {
  View,
  Animated,
  Easing
} from 'react-native';
import Svg, {
  Circle, G
} from 'react-native-svg';

const Face = (props) => {
  const {
    width,
    height,
    radius,
    strokeWidth
  } = props;
  const cx = radius + strokeWidth / 2;
  const cy = cx;
  const faceStyle = {
    position: 'absolute',
  };
  const pulse = new Animated.Value(0)

  const visualAlarm = () => {
    Animated.sequence([
      Animated.timing(
        pulse,
        {
          toValue: 1,
          duration: 250,
          easing: Easing.linear
        }),
      Animated.timing(
        pulse,
        {
          toValue: 0,
          duration: 250,
          easing: Easing.ease
        })
    ]
    ).start(visualAlarm)
  }

  if (props.clock.alarm && (props.app.timerView[0] === 'PAUSE')) {visualAlarm()
  }

  return (
    <Svg
      style={faceStyle}
      width={width}
      height={height}
    >
      <Circle
        cx={cx}
        cy={cy}
        r={radius}
        fillOpacity='1'
        fill='orange'
        stroke='#728c00'
        strokeWidth={strokeWidth}
      />
      <Animated.View style={{opacity: pulse}}>
        <Svg
          style={faceStyle}
          width={width}
          height={height}
        >
          <Circle
            cx={cx}
            cy={cy}
            r={radius}
            fill='red'
            fillOpacity='1'
            stroke='#728c00'
            strokeWidth={strokeWidth}
          />
        </Svg>
      </Animated.View>
    </Svg>
  );
};

export default Face;
//
