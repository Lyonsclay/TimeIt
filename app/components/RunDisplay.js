import React, { Component } from 'react'
import { View, Text, Picker, Animated, Easing, StyleSheet } from 'react-native'

class RunDisplay extends Component {
  constructor(props) {
    super(props)
    let countDown
    this.state = {
      countDown: new Animated.Value(30),
      spin: 30
    }
  }
  componentDidMount() {
    this.state.countDown.setValue(0)
    Animated.timing(
      this.state.countDown,
      {
        toValue: 30,
        duration: 3000
      }
    ).start()
  }

  spinner(n) {
    setTimeout(
      () => { console.log('I do not leak!'); },
      500
    );
   this.setState({ spin: n }) 
  }

  render() {
    const scrollStyle = {
      flex: 4,
      width: 100,
    }
    const numberStyle = {
      fontSize: 28,
      color: '#493d26',
    }
    const Item = Picker.Item
    const minutes = [...Array(60)].map((t, i) => i.toString())
    const AnimatedPicker = Animated.createAnimatedComponent(Picker)

    return (
      <Animated.View>
        <AnimatedPicker
          selectedValue={this.state.spin}
          onValueChange={(n) => this.spinner(n)}
          style={scrollStyle}
          itemStyle={[numberStyle]}
        >
          {minutes.map((n, i) =>
            <Item
              key={i}
              value={i}
              label={n}
            />
           )}
        </AnimatedPicker>
        <Text>{this.state.spin}</Text>
      </Animated.View>
    )
  }
}

export default RunDisplay
