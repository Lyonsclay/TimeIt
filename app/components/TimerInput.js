import React, { Component } from 'react'
import { PickerIOS, View, Text } from 'react-native'

const DurationPicker = (props) => {
  const scrollStyle = {
    flex: 0,
    width: 100,
  }
  const numberStyle = {
    fontSize: 28,
    color: 'orange'
  }
  const Item = PickerIOS.Item
  const minutes = [...Array(60)].map((t, i) => i.toString())
  let time = '0'
  const label = 'Time It'

  return (
      <PickerIOS
        selectedValue={props.clock.duration}
        style={scrollStyle}
        onValueChange={props.setDuration}
      >
        {minutes.map((n, i) =>
          <Item
            key={i}
            value={i}
            label={n}
          />
         )}
      </PickerIOS>
  )
}

const SetTimer = (props) => {
  return (
    <View>
  {(props.app.showTimeSet) ? <DurationPicker props={props} /> : []}
  </View>
)
}

export default DurationPicker
