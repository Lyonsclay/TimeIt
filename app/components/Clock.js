import React, {
  Component
} from 'react'
import {
  connect
} from 'react-redux'
import {
  showTimeSet,
  hideTimeSet
} from '../actions'
import {
  StyleSheet,
  View,
  TouchableOpacity
} from 'react-native';
import Svg, {
  Circle,
  Line
} from 'react-native-svg';
import Face from './Face';
import Hand from './Hand';

const Clock = (props) => {
  const {
    clock,
    screen
  } = props
  const diameter = clock.diameter;
  const radius = diameter / 2;
  const {
    xMid,
    yMid
  } = screen;
  const top = yMid - radius;
  const left = xMid - radius;
  const width = diameter + clock.strokeWidth;
  const height = width;
  const clockStyle = {}

  const _showTimeSet = () => {
    alert(JSON.stringify(props))
    props.showTimeSet()
  }

  return (
    <TouchableOpacity onPress={props.showTimeSet}>
      <View style={styles.container}>
        <Face
          strokeWidth={clock.strokeWidth}
          width={width}
          height={height}
          radius={radius}
        />
        <View style={clockStyle} >
          <Hand
            style={clockStyle}
            strokeWidth={clock.strokeWidth}
            width={width}
            height={height}
            radius={radius}
          />
        </View>
      </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 0
  }
});

/* 
 * const mapStateToProps = (state) => {
 *   return state
 * }
 * const mapDispatchToProps = (dispatch) => {
 *   return {
 *     showTimeSet: () => dispatch(showTimeSet()),
 *     hideTimeSet: () => dispatch(hideTimeSet())
 *   }
 * }*/

export default Clock;
