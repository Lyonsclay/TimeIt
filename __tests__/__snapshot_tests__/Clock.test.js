import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Clock from '../../app/components/Clock.js'
import Face from '../../app/components/Face.js'
import TimerInput from '../../app/components/TimerInput'
import RunDisplay from '../../app/components/RunDisplay'

jest.mock('../../node_modules/react-native-svg/elements/Circle.js', () => 'Circle')
jest.mock('../../node_modules/react-native-svg/elements/Line.js', () => 'Line')

const props = {
  app: {
    mode: ['INIT', 'SELECT', 'RUN', 'FREEZE']
  },
  timer: {
    duration: 30,
    startTime: 0,
    remainder: 30,
  },
  screen: {
    width: 0,
    height: 0
  },
  clock: {
    diameter: 99,
    strokeWidth: 2.5,
  }
}

describe('Clock', () => {
  it('renders', () => {
    const clock = renderer.create(<Clock {...props} />)

    expect(clock).toMatchSnapshot()
  })
})
