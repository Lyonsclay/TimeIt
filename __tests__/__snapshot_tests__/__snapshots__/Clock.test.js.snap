exports[`Clock renders 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 106.5,
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderColor": "red",
          "borderWidth": 1,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "position": "absolute",
          },
          false,
          Object {
            "flex": 0,
            "height": 106.5,
            "width": 106.5,
          },
        ],
      ]
    }>
    <Circle
      cx={53.25}
      cy={53.25}
      fill="orange"
      fillOpacity="1"
      r={52}
      stroke="#728c00"
      strokeWidth={2.5} />
    <View
      style={Object {}}>
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "borderColor": "red",
              "borderWidth": 1,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "position": "absolute",
              },
              false,
              Object {
                "flex": 0,
                "height": 106.5,
                "width": 106.5,
              },
            ],
          ]
        }>
        <Circle
          cx={53.25}
          cy={53.25}
          fill="#e5efe3"
          fillOpacity="1"
          r={52}
          stroke="#728c00"
          strokeWidth={2.5} />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
        ],
      }
    }>
    <View
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "borderColor": "red",
            "borderWidth": 1,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            false,
            Object {
              "flex": 0,
              "height": 104,
              "width": 104,
            },
          ],
        ]
      }>
      <Line
        stroke="brown"
        strokeLinecap="round"
        strokeWidth={5}
        x1={52}
        x2={52}
        y1={10.4}
        y2={52} />
    </View>
  </View>
</View>
`;
